### PR TITLE
Add types for @newswire/frames

### PR DESCRIPTION
**Pull request by Vinnl**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

# Note

The test entry is unchecked - I did run it, but for some reason it's complaining:

> Cannot find module '@newswire/frames' or its corresponding type declarations.

It's not clear to me why that would be the case - I believe I followed the instructions correctly. Possibily related: `dts-gen` didn't create the `tslint.json` for me either; I had to create that myself after the linter complained.

### CODE REVIEW

File: types/newswire__frames/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "projects": [
+        "https://github.com/rdmurphy/frames#readme"
+    ],
+    "devDependencies": {
+        "@types/newswire__frames": "workspace:."
+    },
+    "owners": [
+        {
+            "name": "Vincent",
+            "githubUsername": "DefinitelyTyped"
```
Comment (codershiba) [inline types/newswire__frames/package.json:14]:
```suggestion
            "githubUsername": "Vinnl"
```

**Comment by Vinnl** [comment]
I have no idea what's going wrong here :/

1> ⚠️ A require call resolved to an ESM JavaScript file, which is an error in Node and some bundlers. CommonJS consumers will need to use a dynamic import. https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/CJSResolvesToESM.md
1> 
1> 🎭 Import resolved to a CommonJS type declaration file, but an ESM JavaScript file. https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/FalseCJS.md

As far as I know I haven't used CommonJS modules anywhere, but _something_ is going wrong w.r.t. modules :/

File: types/newswire__frames/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// For the embedded page:
```
Comment (codershiba) [inline types/newswire__frames/index.d.ts:1]:
```suggestion
// For the embedded page:
declare function initFrame(): void;
declare function initFrameAndPoll(delay?: number): void;
declare function sendFrameHeight(height?: number): void;
declare function sendHeightOnLoad(): void;
declare function sendHeightOnResize(): void;
declare function sendHeightOnFramerInit(): void;
declare function sendHeightOnPoll(delay?: number): void;

// For the host: 
declare class Framer {
  constructor(container: HTMLElement, options?: Framer.FramerOptions);
}

declare namespace Framer {
  type FramerOptions = Partial<{
  src: string | null;
  attributes: Record<string, string>;
}>;
}

declare function observeIframe(element: HTMLIFrameElement): (() => void);

declare function autoInitFrames(): void;

export = Framer;
```

**Comment by jakebailey** [comment]
> As far as I know I haven't used CommonJS modules anywhere

Your `package.json` doesn't set `"type": "module"`, so `index.d.ts` is CJS. Upstream sets that, so the types should too, and that's what the check is complaining about. See the linked docs: https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/FalseCJS.md#common-causes

**Comment by Vinnl** [comment]
Thanks Jake. I'm running into other errors now, but since I ended up not using this package, and having to push every time to debug (and not really knowing what's going on), I think I'm going to give up on this :( Sorry!

**Pull request closed without merging**